modules/item/item-wfrp4e.js

```diff
@@ -1592,15 +1593,17 @@ export default class ItemWfrp4e extends Item {
 
     let existing = this.hasCondition(effect.id)
     
-    if (existing && existing.flags.wfrp4e.value == null)
-      return this.deleteEmbeddedDocuments("ActiveEffect", [existing._id])
+    if (existing && existing.flags.wfrp4e.value == null) {
+      await this.deleteEmbeddedDocuments("ActiveEffect", [existing._id])
+      return;
+    }
     else if (existing) {
       await existing.setFlag("wfrp4e", "value", existing.conditionValue - value);
 
       if (existing.flags.wfrp4e.value <= 0)
-        return this.deleteEmbeddedDocuments("ActiveEffect", [existing._id])
+        await this.deleteEmbeddedDocuments("ActiveEffect", [existing._id])
       else
-        return this.updateEmbeddedDocuments("ActiveEffect", [existing])
+        await this.updateEmbeddedDocuments("ActiveEffect", [existing])
```

Comment:
I don't see the reason here, returning a promise is intended

Reply:
preference, i can revert this.